### PR TITLE
Fix http and subcontrol

### PR DESCRIPTION
**Pull request by minhaj10p**

### CODE REVIEW

File: generator/generator_test.go [hunk 1]
```diff
@@ -156,6 +156,44 @@ func TestCreateCatalogsFromProfileWithBadHref(t *testing.T) {
...
 }
 
+func TestSubControlsMapping(t *testing.T) {
+
+	profile := profile.Profile{
+		Imports: []profile.Import{
+			profile.Import{
+				Href: &catalog.Href{
+					URL: func() *url.URL {
+						url, _ := url.Parse("/home/minhaj/go/src/github.com/minhaj10p/cgen/NIST_SP-800-53_rev4_catalog.json")
```
Comment (mohuk) [inline generator/generator_test.go]: Trying to parse absolute url `home/minhaj/go/src/github.com/minhaj10p/...`, wouldn't this fail on other machines ?

File: generator/intersection.go [hunk 2]
```diff
@@ -41,8 +43,47 @@ func CreateCatalogsFromProfile(profile *profile.Profile) []*catalog.Catalog {
...
+					mappedCatalogs.Groups[i].Controls[j].Subcontrols = append(mappedCatalogs.Groups[i].Controls[j].Subcontrols, catalog.Subcontrol{
+						Id: call.SubcontrolId,
+					})
+				}
+			}
+		}
+	}
+
+}
+func isSUbControl(s string) bool {
```
Comment (mohuk) [inline generator/intersection.go]: Typo: `isSUbControl` control should be `is SubControl`

**Review by mohuk — CHANGES_REQUESTED** [review]

File: generator/generator_test.go (hunk 1 above)
Comment (minhaj10p) [reply generator/generator_test.go]: yes, I feel stupid.
can you suggest something else or should I remove this test then?

Comment (mohuk) [reply generator/generator_test.go]: can we switch it to relative paths ?

Comment (minhaj10p) [reply generator/generator_test.go]: sure!

**Pull request closed without merging**